unicore/comments/service/models.py

```diff
@@ -50,6 +53,12 @@ def _pk_expression(cls, data):
     def pk_expression(self):
         return self.__class__._pk_expression(self.row_dict)
 
+    @classmethod
+    def match_all_expression(cls, **kwargs):
+        expression = [cls.__table__.c[k] == v
+                      for k, v in kwargs.iteritems()]
+        return and_(*expression)
+
     def to_dict(self):
```

Comment:
The semantics of `__dict__` and `to_dict` differ by design. `to_dict` returns a copy and it explicitly includes only fields that the model "exports" - by default only database columns.

I like using the standard `__dict__` implementation for introspection, but if there's a good use case for overriding it...